### PR TITLE
Create a servlet that handles voting on reviews

### PR DESCRIPTION
**Pull request by ccjeane**
- Datastore now contains information for total count, positive and negative reviews. 
- When users click on the upvote/downvote button, this servlet will be called and the values will be updated accordingly.

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/VotingServlet.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    Query<Entity> query = Query.newEntityQueryBuilder()
+      .setKind("Review")
+      .setFilter(PropertyFilter.eq("id", id));
+
+    PreparedQuery results = datastore.prepare(query);
+
+    // Check to make sure the datastore returned something
+    if (results[0] != null){
+      Entity review = results[0];
+      int total = (int) entity.getProperty("total");
```
Comment (kcirerick) [inline src/main/java/com/google/sps/servlets/VotingServlet.java:39]: no need to retrieve this value, you don't use it again

Comment (kcirerick) [inline src/main/java/com/google/sps/servlets/VotingServlet.java:39]: where did you initialize entity? did you mean review?

File: src/main/java/com/google/sps/servlets/VotingServlet.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        positive++;
+      } else {
+        review.setProperty("negative", negative + 1);
+        negative++;
+      }
+
+      review.setProperty("total", positive - negative);
+
+      // Adds the review back to the Datastore
+      datastore.put(review);
```
Comment (kcirerick) [inline src/main/java/com/google/sps/servlets/VotingServlet.java:53]: Do you need to add it back to update it? Should you delete it at some point beforehand then?

File: src/main/java/com/google/sps/servlets/ReviewServlet.java [hunk 3]
```diff
@@ -33,7 +33,10 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       long id = entity.getKey().getId();
       String message = (String) entity.getProperty("message");
       Date timestamp = (Date) entity.getProperty("date");
-      reviews.add(message + " - " + timestamp.toString());
+      int total = (int) entity.getProperty("total");
+      int positive = (int) entity.getProperty("positive");
+      int negative = (int) entity.getProperty("negative");
```
Comment (kcirerick) [inline src/main/java/com/google/sps/servlets/ReviewServlet.java:38]: you don't use positive or negative inside of the loop

File: src/main/java/com/google/sps/servlets/VotingServlet.java (hunk 1 above)
Comment (ccjeane) [reply src/main/java/com/google/sps/servlets/VotingServlet.java:39]: Yes

File: src/main/java/com/google/sps/servlets/VotingServlet.java (hunk 2 above)
Comment (ccjeane) [reply src/main/java/com/google/sps/servlets/VotingServlet.java:53]: Adding it back replaces the old one / updates the system.

File: src/main/java/com/google/sps/servlets/ReviewServlet.java (hunk 3 above)
Comment (ccjeane) [reply src/main/java/com/google/sps/servlets/ReviewServlet.java]: Good catch - fixed

File: src/main/java/com/google/sps/servlets/VotingServlet.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+    // Find the review that corresponds to the given id
+    Query<Entity> query = Query.newEntityQueryBuilder()
+      .setKind("Review")
+      .setFilter(PropertyFilter.eq("id", id));
+
+    PreparedQuery results = datastore.prepare(query);
+
+    // Check to make sure the datastore returned something
+    if (results[0] != null){
```
Comment (dterr) [inline src/main/java/com/google/sps/servlets/VotingServlet.java:37]: ### nit
Whitespace before ```{```

**Pull request merged**